sdk_entrepot_gpf/workflow/action/UploadAction.py

```diff
@@ -127,6 +144,18 @@ def __add_tags(self) -> None:
             self.__upload.api_add_tags(self.__dataset.tags)
             Config().om.info(f"Livraison {self.__upload['name']} : les {len(self.__dataset.tags)} tags ont été ajoutés avec succès.")
 
+    def __add_carte_tags(self, upload_step) -> None:
+        """En mode cartes.gouv, ajoute les tags nécessaires."""
+        # lister toutes les clés dans la section compatibility_cartes, filtrer chaque clé qui commence par upload_step,
+        # mettre la fin de la clé dans un tag et mettre la value (en string) comme value du tag (self.__upload.api_add_tags(...))
+        d_section = Config().get_config()["compatibility_cartes"]
+        if self.__upload is not None and d_section is not None:
+            for s_key, s_val in d_section.items():
+                if s_key.startswith(upload_step):
+                    d_tag = {}
+                    d_tag[s_key[len(upload_step) + 1 :]] = str(s_val)
+                    self.__upload.api_add_tags(d_tag)
```

Comment:
Il faudrait mieux créer le dictionnaire vide avant la boucle `for` et appeler une seule fois `self.__upload.api_add_tags` à la fin si le dictionnaire n'est pas vide.

ca serait bien d'avoir un petit commentaire pour expliquer rapidement la logique derrière le `s_key[len(upload_step) + 1 :]` pour qu'on comprenne directement à la relecture